packages/passport/sdk/src/Passport.ts

```diff
@@ -299,4 +302,55 @@ export class Passport {
     const getUserInfoResult = await this.multiRollupApiClients.passportProfileApi.getUserInfo({ headers });
     return getUserInfoResult.data.linked_addresses;
   }
+
+  public async linkExternalWallet(params: LinkWalletParams): Promise<LinkedWallet> {
+    track('passport', 'linkWallet', { type: params.type });
+
+    const user = await this.authManager.getUser();
+    if (!user) {
+      throw new PassportError('User is not logged in', PassportErrorType.NOT_LOGGED_IN_ERROR);
+    }
+
+    const isRegisteredWithIMX = isUserImx(user);
+    const isRegisteredWithZkEvm = isUserZkEvm(user);
+    if (!isRegisteredWithIMX && !isRegisteredWithZkEvm) {
+      throw new PassportError('User has not been registered', PassportErrorType.USER_NOT_REGISTERED_ERROR);
+    }
+
+    const headers = { Authorization: `Bearer ${user.accessToken}` };
+    const linkWalletV2Request = {
+      type: params.type,
+      wallet_address: params.walletAddress,
+      signature: params.signature,
+      nonce: params.nonce,
+    };
+
+    return await withPassportError(async () => {
+      const linkWalletV2Result = await this.multiRollupApiClients
+        .passportProfileApi.linkWalletV2({ linkWalletV2Request }, { headers });
+
+      // Throw PassportError for Bad Request.
+      if (linkWalletV2Result.status === 400) {
```

Comment:
According to [the docs](https://docs.immutable.com/api/zkevm/reference/#/operations/link-wallet-v2) this endpoint can return a `400`, `401`, `403`, or a `500`. Can we invert the logic so that we're handling these status codes please? It may also be good to log the status code if it's something other than an APIError, e.g:
```
if (linkWalletV2Result.status === 200) {
  return linkWalletV2Result.data;
} else {
  if (isAPIError(linkWalletV2Result.data)) {
    const { code, message } = linkWalletV2Result.data;
    switch (code) {
      ...
    }
  } else {
    throw new PassportError(`Link wallet request failed with status code ${linkWalletV2Result.status}`, PassportErrorType.LINK_WALLET_GENERIC_ERROR);
  }
}
```